### PR TITLE
 posix: add missing defines in poll.h

### PR DESCRIPTION
**Pull request by msobkowski**
This PR adds a couple of missing POSIX defines in `poll.h`.

### CODE REVIEW

File: include/zephyr/posix/netinet/in.h [hunk 1]
```diff
@@ -17,6 +17,29 @@ extern "C" {
...
+&& ((const struct in6_addr *)addr)->s6_addr32[2] == 0         \
+&& ((const struct in6_addr *)addr)->s6_addr32[3] == htonl(1))
+
+#define IN6_IS_ADDR_LINKLOCAL(addr) \
+(((const struct in6_addr *)addr)->s6_addr32[0] & htonl(0xffc) == htonl(0xfe8))
+
+#define IN6_IS_ADDR_SITELOCAL(addr) \
+(((const struct in6_addr *)addr)->s6_addr32[0] & htonl(0xffc) == htonl(0xfec))
+
+#define IN6_IS_ADDR_V4MAPPED(addr) \
```
Comment (jukkar) [inline include/zephyr/posix/netinet/in.h:38]: I recently added these to `include/zephyr/net/socket.h` by commit 8f49925a0fcc

File: include/zephyr/posix/netinet/in.h [hunk 2]
```diff
@@ -40,6 +40,8 @@ typedef uint32_t in_addr_t;
 && ((const struct in6_addr *)addr)->s6_addr32[1] == 0         \
 && ((const struct in6_addr *)addr)->s6_addr32[2] == 0xffff)   \
 
+#define INADDR_BROADCAST ((in_addr_t)0xffffffff)
```
Comment (jukkar) [inline include/zephyr/posix/netinet/in.h:43]: Already present in `include/zephyr/net/net_ip.h` by commit 606eeed8c92f

File: include/zephyr/posix/netinet/in.h (hunk 1 above)
Comment (msobkowski) [reply include/zephyr/posix/netinet/in.h]: Oh, I missed that, thanks. Removed it from this PR.

File: include/zephyr/posix/netinet/in.h (hunk 2 above)
Comment (msobkowski) [reply include/zephyr/posix/netinet/in.h]: Dropped.

**Review by cfriedt — APPROVED** [review]

**Comment by cfriedt** [comment]
@msobkowski - I wouldn't argue if you also wanted to define POLLRDNORM, POLLRDBAND, 
POLLWRNORM, and POLLWRBAND

**Review by kartben — APPROVED** [review]

**Pull request merged**